src/test/daml/Daml/Finance/Instrument/Bond/Test/Util.daml

```diff
@@ -51,15 +54,46 @@ originateCallableBond depository issuer label description observers lastEventTim
       couponSchedule = createPaymentPeriodicSchedule firstCouponDate holidayCalendarIds
         businessDayConvention couponPeriod couponPeriodMultiplier issueDate maturityDate
       callSchedule = couponSchedule
+
+    -- Create a floating rate bond factory
+    callableBondFactoryCid <- toInterfaceContractId @Callable.F <$> submit issuer do
+      createCmd Callable.Factory with
+        provider = issuer
+        observers = M.empty
+
     -- CREATE_CALLABLE_BOND_INSTRUMENT_BEGIN
-    cid <- toInterfaceContractId <$> submitMulti [depository, issuer] [] do
-      createCmd Callable.Instrument with
-        depository; issuer; id = Id label; version = "0"; description
-        observers = M.fromList observers; lastEventTimestamp; prevEvents = []; couponSchedule
-        callSchedule; noticeDays; holidayCalendarIds; calendarDataProvider; dayCountConvention
-        useAdjustedDatesForDcf; floatingRate; couponRate; capRate; floorRate; currency; notional
+    let
+      instrumentId = Id label
```

Comment:
as it is only used once, I would not define the instrumentId but rather use the value inline. same goes for the next line.